### ISSUE TITLE
Need a script to automate MinGW-based HX DOS extender releases

### ISSUE THREAD

**Issue by joncampbell123** [joncampbell123/dosbox-x#591]
Needs to include compiled binary, HDPMI32, HX DOS and HXGUI binaries, and a copy of MSVCRT.DLL known to work taken from an older Windows system.

**Comment by yksoft1** [comment]
You may also need CRTDLL.DLL.
(There is a modified HX DOS distribution that contains an OpenGL driver for Voodoo, but I don't think Dosbox-X can be used practically on a PC that still compatible with Voodoo card).

**Comment by joncampbell123** [comment]
The copy of MSVCRT.DLL I used from Windows 98 did not need CRTDLL.DLL, neither did the one from Windows XP.

**Comment by Wengier** [comment]
@joncampbell123 I have already made the HX-DOS build script fully automated by including all **required** HX DOS Extender files and MSVCRT.DLL (in /build-scripts/mingw/hxdos) in my latest pull request #1666. The updated build script (make-mingw-hx-dos.sh) will also automatically run PESTUB.EXE to patch the HX-DOS build's dosbox-x.exe file before zipping it so that the program DPMILD32.EXE is no longer needed for running it in DOS.

Therefore it becomes extremely easy to run the HX-DOS build: simply run dosbox-x.exe directly in a real DOS system after unzipping, as the package has already taken care of everything needed.

BTW, you may want to slightly modify the directories defined by environment variables $copydir and $hxdir in make-mingw-hx-dos.sh in case the defined one does not match the actual build environment on your computer. They are defined as follows (as in the repository):

copydir="build-scripts/mingw/dosbox-x-mingw-hx-dos"
hxdir="build-scripts/mingw/hxdos"

**Comment by Wengier** [comment]
@joncampbell123 I noticed that DOSBox-X 0.83.3 is released, but HX DOS binaries are not yet included in the HX-DOS version. Is there any issue with the current build script? Please let me know if you find any.

**Comment by joncampbell123** [comment]
Now that I think about it, I have an older version of the script one level up from the source directory that I ran. Let me try again.

**Comment by joncampbell123** [comment]
It looks like the script needs a little more work, which can then be done in the next release.

**Comment by Wengier** [comment]
@joncampbell123 You can try the build script for the Windows installer in the next release too. This build script (build_windows_installer.bat in the repo root) is also fully automated. Right now there are 3 separate zip files for the Windows platform (I.e. VS, MinGW32, MinGW64), and by looking at the file names many Windows users seem to simply ignore the MinGW builds, but the Windows installer let you install all builds in one package and select a default build to run, so it will become very convenient for them.

**Comment by Wengier** [comment]
@joncampbell123 All these build scripts are primarily for convenience. You will decide how to use them of course. In case you find any issue(s) just let me know.

**Comment by joncampbell123** [comment]
@Wengier I'll see what I can do with them, thank you.

**Comment by Wengier** [comment]
@joncampbell123 I think I can feel you have some concerns for this script (possibly related to the Windows platform), which is understandable. So I have been following what we already have, and nothing else. The recent change to INSTALL was an attempt to solve possible confusion with this documentation file as seen in Issue #1707, and it has nothing to do with this.
